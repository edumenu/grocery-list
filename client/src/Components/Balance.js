import React, { useContext } from 'react'
import { GlobalContext } from '../context/GlobalContext';

function Balance() {
    const { groceryLists } = useContext(GlobalContext);
    const amount = groceryLists.map(groceryList => groceryList.amount)
    const total = amount.reduce((previousAmount, currentAmount) => (previousAmount += currentAmount), 0).toFixed(2)

    return (
        <div className="p-4 lg:w-1/3 md:w-1/3 sm:w-full w-full">
            <div className="card_element4 px-2 py-6 rounded-lg">
                <h2 data-testid="h3tag" className="font-medium text-3xl lg:text-2xl text-gray-300">${total}</h2>
                <p className="leading-relaxed text-gray-300">Your balance</p>
            </div>
        </div>
    )
}

export default Balance
