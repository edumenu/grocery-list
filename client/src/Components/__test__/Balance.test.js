import React from 'react';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Balance from '../Balance';

afterEach(cleanup);

it('renders', () => {
    const { asFragment } = render(<Balance />);
    expect(asFragment()).toMatchSnapshot();
});

if("Finds the balance", () => {
    const { getByTestId, getByText } = render(<Balance />);

    expect(getByTestId("h3tag")).toHaveTextContent(300);
    
});