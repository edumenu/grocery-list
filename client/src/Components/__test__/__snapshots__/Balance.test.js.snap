// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="p-4 lg:w-1/3 md:w-1/3 sm:w-full w-full"
  >
    <div
      class="card_element4 px-2 py-6 rounded-lg"
    >
      <h2
        class="font-medium text-3xl text-gray-300"
        data-testid="h3tag"
      >
        $0.00
      </h2>
      <p
        class="leading-relaxed text-gray-300"
      >
        Your balance
      </p>
    </div>
  </div>
</DocumentFragment>
`;
